sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/PlanAdaptiveDynamicPruningFilters.scala

```diff
@@ -62,8 +63,18 @@ case class PlanAdaptiveDynamicPruningFilters(
           val broadcastValues = SubqueryBroadcastExec(
             name, index, buildKeys, newAdaptivePlan)
           DynamicPruningExpression(InSubqueryExec(value, broadcastValues, exprId))
-        } else {
+        } else if (onlyInBroadcast) {
           DynamicPruningExpression(Literal.TrueLiteral)
+        } else {
+          // we need to apply an aggregate on the buildPlan in order to be column pruned
+          val alias = Alias(buildKeys(index), buildKeys(index).toString)()
+          val aggregate = Aggregate(Seq(alias), Seq(alias), buildPlan)
+
+          val session = adaptivePlan.context.session
+          val sparkPlan = session.sessionState.planner.plan(ReturnAnswer(aggregate)).next()
```

Comment:
Here we can not use the `QueryExecution.prepareExecutedPlan()` method, which will cause the wholestage code gen be inserted twice.

Reply:
ah, then how about go back to `QueryExecution.createSparkPlan`?

Reply:
and leave a comment to explain it.

Reply:
Done